### PR TITLE
Make project compatible with Gradle 7

### PR DESCRIPTION
**Pull request by ValentinModan**
- use implementation, testImplementation instead of compile and testCompile
- upgrade gradle version to 6.9

Change-Id: Id7b7a362348ed7214095251e825d46babce5d67a

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https://services.gradle.org/distributions/gradle-4.9-all.zip
+distributionUrl=https://services.gradle.org/distributions/gradle-6.9-all.zip
```
Comment (fioan89) [inline gradle/wrapper/gradle-wrapper.properties]: There should be some other files changed, please use `./gradlew wrapper --gradle-version 6.9` to upgrade

File: build.gradle [hunk 2]
```diff
@@ -28,20 +28,20 @@ task sourcesJar(type: Jar, dependsOn: classes) {
...
+    testImplementation(
             'junit:junit:4.12',
             'org.spockframework:spock-core:1.1-groovy-2.4',
             'org.mockito:mockito-core:2.8.47')
 }
 
 configurations.all {
     resolutionStrategy {
-        force 'org.codehaus.groovy:groovy-all:2.4.12'
+        force 'org.codehaus.groovy:groovy-all:2.5.12'
```
Comment (fioan89) [inline build.gradle]: I have to double check it but I think groovy 3 is required for gradle 7 compatibility

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (ValentinModan) [reply gradle/wrapper/gradle-wrapper.properties]: Done.

File: build.gradle (hunk 2 above)
Comment (ValentinModan) [reply build.gradle]: Done.

**Review by fioan89 — APPROVED** [review]

**Pull request merged**